src/dmd/dsymbolsem.d

```diff
@@ -4851,6 +4874,92 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
+                Loc dtloc = cldec.dtor.getLoc();
+
+                StorageClass stc = STC.safe | STC.nothrow_ | STC.pure_ | STC.nogc;
+                stc = mergeFuncAttrs(stc, cldec.dtor);
+
+                DtorDeclaration deldtor;
+
+                if (!cldec.dtor.fbody)
+                {
+                    // if `cldec.dtor` is an extern prototype, then we should also extern to the deleting destructor
+                    deldtor = new DtorDeclaration(dtloc, Loc.initial, STC.extern_ | stc, Id.__cpp_deleting_dtor);
+                }
+                else
+                {
+                    // create deleting destructor, and assign it to vtbl
+
+                    // we need to extern to `operator delete`
+//                    FuncDeclaration cpp_delete;
+//                    __gshared FuncDeclaration cpp_delete;
+//                    if (!cpp_delete)
+//                    {
+//                        auto parameters = new Parameters();
+//                        parameters.push(new Parameter(0, Type.tvoidptr, Id.p, null));
+//                        auto tf = new TypeFunction(parameters, Type.tvoid, 0, LINK.cpp);
+//                        cpp_delete = new FuncDeclaration(Loc.initial, Loc.initial, Id.__cpp_operator_delete, STC.static_ | STC.extern_ | STC.nothrow_ | STC.trusted | STC.nogc, tf);
+//
+////                        cldec.dtors.push(deldtor);
+////                        cldec.members.push(deldtor);
+//                        cpp_delete.dsymbolSemantic(sc2);
+//                    }
+
+                    // create a function that calls the destructor, followed by `operator delete`
+
+                    // call destructor
+                    Expression dtor_exp;
+                    dtor_exp = new ThisExp(dtloc);
+                    dtor_exp = new DotVarExp(dtloc, dtor_exp, cldec.dtor, false);
+                    dtor_exp = new CallExp(dtloc, dtor_exp);
+
+                    // call operator delete
+//                    Expression delete_exp;
+//                    delete_exp = new ThisExp(dtloc);
+//                    delete_exp = new CastExp(dtloc, delete_exp, Type.tvoidptr);
+//                    if (stc & STC.safe)
+//                        stc = (stc & ~STC.safe) | STC.trusted;
+//                    delete_exp = new CallExp(dtloc, cpp_delete, delete_exp);
+
+                    // make it into a destructor
+                    deldtor = new DtorDeclaration(dtloc, Loc.initial, stc, Id.__cpp_deleting_dtor);
+                    deldtor.generated = true;
+                    deldtor.storage_class |= STC.inference;
+                    deldtor.fbody = new ExpStatement(dtloc, dtor_exp);
+//                    deldtor.fbody = new ExpStatement(dtloc, Expression.combine(dtor_exp, delete_exp));
+                }
+
+                // add it to the class and run semantic
+                cldec.dtors.push(deldtor);
+                cldec.members.push(deldtor);
+                deldtor.dsymbolSemantic(sc2);
```

Comment:
Just FYI, someone might have more better idea here...  
`dsymbolSemantic` goes all wrong if I give it any hint that this function is virtual/override when it runs. It can't deal with reserved vtable slots like this here. The simplest solution seemed to fix up the relevant state (immediately below) after the function processes as if a normal non-virtual method. I didn't want to mess with the chaos in `dsymbolSemantic` to try and teach it about reserved vtable slots. And that would also add extra state to node structures too... so this hack(-ish) just below... looks okay?